src/nvim/option.c

```diff
@@ -1788,6 +1795,21 @@ do_set (
             /* If error detected, print the error message. */
             if (errmsg != NULL)
               goto skip;
+            if (saved_origval != NULL) {
+              char_u buf_type[7];
+
+              vim_snprintf((char *)buf_type, 7, "%s",
```

Comment:
It is better to use `ARRAY_SIZE(buf_type)` in place of 7, even though replacing `snprintf` with `vim_snprintf` will silence these warnings. Also I would suggest using `char buf_type[7]`: in any case there will be one cast, but cast to `char_u` with “proper” base type is better then cast to `char` with deprecated type.